python/chemiscope/structures/_stk.py

```diff
@@ -0,0 +1,198 @@
+from collections import Counter
+
+try:
+    import stk
+
+    HAVE_STK = True
+except ImportError:
+    HAVE_STK = False
+
+
+def _stk_valid_structures(
+    frames: stk.Molecule | list[stk.Molecule],
+) -> tuple[list[stk.Molecule], bool]:
+    if HAVE_STK and isinstance(frames, stk.Molecule):
+        # deal with the user passing a single frame
+        return [frames], True
+    elif HAVE_STK and isinstance(frames[0], stk.Molecule):
+        for frame in frames:
+            assert isinstance(frame, stk.Molecule)
+        return frames, True
+    else:
+        return frames, False
+
+
+def _stk_to_json(molecule: stk.Molecule) -> dict[str : int | list]:
+    """Implementation of frame_to_json for stk.Molcule.
+
+    The current implementation assumes no periodic information, which is safe
+    for the majority of stk molecules. If necessary, we can add cell information
+    in the future.
+
+    """
+    pos_mat = molecule.get_position_matrix()
+    data = {}
+    data["size"] = molecule.get_num_atoms()
+    data["names"] = [atom.__class__.__name__ for atom in molecule.get_atoms()]
+    data["x"] = [float(pos_mat[atom.get_id()][0]) for atom in molecule.get_atoms()]
+    data["y"] = [float(pos_mat[atom.get_id()][1]) for atom in molecule.get_atoms()]
+    data["z"] = [float(pos_mat[atom.get_id()][2]) for atom in molecule.get_atoms()]
+
+    return data
+
+
+def _stk_all_atomic_environments(frames, cutoff):
+    "Extract all atomic environments out of a set of ASE Atoms objects"
+    environments = []
+    for structure_i, frame in enumerate(frames):
+        for atom in frame.get_atoms():
+            environments.append((structure_i, atom.get_id(), cutoff))
+
+    return environments
+
+
+def _stk_composition_properties(frames, environments=None):
```

Comment:
TBH, we might want to remove this, now that we have better ways of handling additional properties. I'll look into this in a separate PR!

Reply:
Considering it is basically copy and paste from the ase functions, I will leave it in. I was not certain of its purpose but went for as many of the same features as ASE as I could in the shortest time